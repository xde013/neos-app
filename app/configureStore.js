/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import createSagaMiddleware from 'redux-saga';
import createReducer from './reducers';

export default function configureStore(initialState = {}, history) {
  let composeEnhancers = compose;
  const reduxSagaMonitorOptions = {};

  /* istanbul ignore next */
  if (process.env.NODE_ENV !== 'production' && typeof window === 'object') {
    /* eslint-disable no-underscore-dangle */
    if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({});
    /* eslint-enable */
  }

  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with two middlewares
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const enhancers = [applyMiddleware(...middlewares)];

  const store = createStore(
    createReducer(),
    initialState,
    composeEnhancers(...enhancers),
  );

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectedReducers = {}; // Reducer registry
  store.injectedSagas = {}; // Saga registry

  // Hot Reloadable Reducers
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createReducer(store.injectedReducers));
    });
  }

  return store;
}
