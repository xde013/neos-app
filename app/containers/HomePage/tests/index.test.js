/* eslint-disable react/jsx-props-no-spreading */
/**
 * Home Test
 */
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { HomePage, mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';

describe('<HomePage />', () => {
  let store;
  let props;
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  beforeEach(() => {
    props = {
      loading: false,
      error: false,
      loadNeos: jest.fn(),
      neos: [],
    };
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <HomePage {...props} />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <HomePage {...props} />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch the neos on mount', () => {
    const loadNeosSpy = jest.fn();

    render(
      <Provider store={store}>
        <HomePage {...{ ...props, loadNeos: loadNeosSpy }} />
      </Provider>,
    );
    expect(loadNeosSpy).toHaveBeenCalled();
  });

  describe('Props', () => {
    describe('loadNeos', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.loadNeos).toBeDefined();
      });
    });
  });
});
