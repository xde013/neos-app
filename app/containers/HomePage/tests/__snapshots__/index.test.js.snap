// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  style="margin-bottom: 50px;"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-pressed="true"
          class="MuiButtonBase-root MuiToggleButton-root Mui-selected"
          color="primary"
          tabindex="0"
          title="Toggle to Table View"
          type="button"
          value="false"
        >
          <span
            class="MuiToggleButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h15c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 2v3H5V5h15zm-5 14h-5v-9h5v9zM5 10h3v9H5v-9zm12 9v-9h3v9h-3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Discover NEOs | 
          <em>
             Near Earth Objects 
          </em>
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <div
          class="MuiFormControl-root makeStyles-formControl-524"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-placeholder-label-label demo-simple-select-placeholder-label"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="demo-simple-select-placeholder-label"
              role="button"
              tabindex="0"
            >
              <em>
                All
              </em>
            </div>
            <input
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-578 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-579"
                style="width: 0.01px;"
              >
                <span>
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-contained"
          >
            Filter by Orbit
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    style="opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="LoadingIndicator__Wrapper-c3cbkl-0 cnqvsL"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    hidden=""
    style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
  >
    <div
      data-testid="table"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
        style="position: relative;"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MTableToolbar-root-632 MuiToolbar-gutters"
        >
          <div
            class="MTableToolbar-title-636"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              NEOs Diameters
            </h6>
          </div>
          <div
            class="MTableToolbar-spacer-634"
          />
          <div
            class="MuiFormControl-root MuiTextField-root MTableToolbar-searchField-637"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd"
            >
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionStart"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  role="presentation"
                  title="Search"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
                placeholder="Search"
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="MTableToolbar-actions-635"
          >
            <div>
              <div>
                <span>
                  <button
                    aria-label="Export"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                    tabindex="0"
                    title="Export"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 12v7H5v-7H3v7c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zm-6 .67l2.59-2.58L17 11.5l-5 5-5-5 1.41-1.41L11 12.67V3h2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </span>
                <span />
              </div>
            </div>
          </div>
        </div>
        <div
          style="overflow-x: auto;"
        >
          <div>
            <div
              style="overflow-y: auto;"
            >
              <table
                class="MuiTable-root"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-676 MuiTableCell-alignLeft"
                      scope="col"
                      style="background-color: rgb(11, 61, 145); color: rgb(255, 255, 255);"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-roledescription="Draggable item. Press space bar to lift"
                          data-react-beautiful-dnd-drag-handle="1"
                          data-react-beautiful-dnd-draggable="1"
                          draggable="false"
                          tabindex="0"
                        >
                          Name
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-676 MuiTableCell-alignRight"
                      scope="col"
                      style="background-color: rgb(11, 61, 145); color: rgb(255, 255, 255);"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-roledescription="Draggable item. Press space bar to lift"
                          data-react-beautiful-dnd-drag-handle="1"
                          data-react-beautiful-dnd-draggable="1"
                          draggable="false"
                          tabindex="0"
                        >
                          Min Estimated Diameter (km)
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-676 MuiTableCell-alignRight"
                      scope="col"
                      style="background-color: rgb(11, 61, 145); color: rgb(255, 255, 255);"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-roledescription="Draggable item. Press space bar to lift"
                          data-react-beautiful-dnd-drag-handle="1"
                          data-react-beautiful-dnd-draggable="1"
                          draggable="false"
                          tabindex="0"
                        >
                          Max Estimated Diameter (km)
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                    style="height: 245px;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body"
                      colspan="3"
                      style="padding-top: 0px; padding-bottom: 0px; text-align: center;"
                    >
                      No records to display
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <table
          class="MuiTable-root"
        >
          <tfoot
            class="MuiTableFooter-root"
            style="display: grid;"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-footer"
            >
              <td
                class="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root Component-paginationRoot-600"
                colspan="3"
                icons="[object Object]"
                style="float: right; overflow-x: auto;"
              >
                <div
                  class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar Component-paginationToolbar-601 MuiToolbar-gutters"
                >
                  <div
                    class="MuiTablePagination-spacer"
                  />
                  <p
                    class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-602 MuiTypography-body2 MuiTypography-colorInherit"
                  >
                    Rows per page:
                  </p>
                  <div
                    class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot Component-paginationSelectRoot-603"
                  >
                    <div
                      aria-haspopup="listbox"
                      aria-labelledby=" "
                      class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                      role="button"
                      tabindex="0"
                    >
                      <div
                        style="padding: 0px 5px;"
                      >
                        5 rows 
                      </div>
                    </div>
                    <input
                      type="hidden"
                      value="5"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                  <p
                    class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-602 MuiTypography-body2 MuiTypography-colorInherit"
                  >
                    0-0 of 0
                  </p>
                  <div
                    class="MTablePaginationInner-root-712"
                  >
                    <span
                      class=""
                      title="First Page"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                            />
                            <path
                              d="M24 24H0V0h24v24z"
                              fill="none"
                            />
                          </svg>
                        </span>
                      </button>
                    </span>
                    <span
                      class=""
                      title="Previous Page"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                            />
                          </svg>
                        </span>
                      </button>
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-caption"
                      style="flex: 1; text-align: center; align-self: center; flex-basis: inherit;"
                    >
                      1-0 of 0
                    </span>
                    <span
                      class=""
                      title="Next Page"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                            />
                          </svg>
                        </span>
                      </button>
                    </span>
                    <span
                      class=""
                      title="Last Page"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                            />
                          </svg>
                        </span>
                      </button>
                    </span>
                  </div>
                </div>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
      <a
        download="my_neos_data.csv"
        hidden=""
        href="#!"
      >
        Export
      </a>
    </div>
  </div>
</div>
`;
